src/eddymotion/model/base.py

```diff
@@ -448,6 +449,69 @@ class DKIModel(BaseModel):
     _model_class = "dipy.reconst.dki.DiffusionKurtosisModel"
 
 
+class GaussianProcessModel:
+    """A Gaussian process model for DWI data based on [Andersson15]_."""
+
+    __slots__ = (
+        "_dwi",
+        "_kernel",
+        "_num_iterations",
+        "_gpr",
+    )
+
+    def __init__(self, dwi, kernel, **kwargs):
+        """Implement object initialization.
+
+        Parameters
+        ----------
+        dwi : :obj:`~eddymotion.dmri.DWI`
+            The DWI data.
```

Comment:
Do we need this dwi argument? It seems it is not used

Reply:
Not used yet, but the way I see things we would give a `DWI` instance to the class, which would compute the kernels with the gradient table, and we would give it an index for the fit/predict part. Otherwise, do you think this should be the estimator's task?

Reply:
Ha, no I see we had already discussed this. I think this is only necessary in fit()